pennylane/math/multi_dispatch.py

```diff
@@ -353,7 +354,7 @@ def dot(tensor1, tensor2, like=None):
     if like in {"tensorflow", "autograd"}:
         shape_y = len(np.shape(y))
         shape_x = len(np.shape(x))

```

Comment:
Could consider renaming those variables to `ndim_y` and `ndim_x` because they describe the number of dimensions, not the shape :) Not important, though!